boards/arm/ip_k66f/doc/index.rst

```diff
@@ -0,0 +1,168 @@
+- On-board debug probe J-Link-OB for programming
+
+.. image:: ./ip_k66f.png
+   :width: 720px
+   :align: center
+   :alt: IP-K66F
+
+Hardware
+********
+
+- MK66FN2M0VMD18 MCU (180 MHz, 2 MB flash memory, 256 KB RAM, low-power,
+  crystal-less USB
+- Dual role USB interface with micro-B USB connector
+- 2 User LED
+- On-board debug probe J-Link-OB for programming
+- Micrel/Microchip Ethernet Switch KSZ8794CNX with 3 RJ45 connectors
+
+For more information about the K66F SoC and IP-K66F board:
+
+- `K66F Website`_
+- `K66F Datasheet`_
+- `K66F Reference Manual`_
+- `IP-K66F Website`_
+- `IP-K66F User Guide`_
+- `IP-K66F Schematics`_
+
+Supported Features
+==================
+
+The ip_k66f board configuration supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| SYSTICK   | on-chip    | systick                             |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| USB       | on-chip    | USB device                          |
++-----------+------------+-------------------------------------+
+| WATCHDOG  | on-chip    | watchdog                            |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | soc flash                           |
++-----------+------------+-------------------------------------+
+| RTC       | on-chip    | rtc                                 |
++-----------+------------+-------------------------------------+
+
+The default configuration can be found in the defconfig file:
+
+	``boards/arm/ip_k66f/ip_k66f_defconfig``
+
+Micrel/Microchip KSZ8794CNX Ethernet Switch is not currently
+supported.
+
+Connections and IOs
+===================
+
+The K66F SoC has five pairs of pinmux/gpio controllers.
+
++-------+-----------------+---------------------------+
+| Name  | Function        | Usage                     |
++=======+=================+===========================+
+| PTA8  | GPIO            | Red LED                   |
++-------+-----------------+---------------------------+
+| PTA10 | GPIO            | RED LED                   |
++-------+-----------------+---------------------------+
+
+System Clock
+============
+
+The K66F SoC is configured to use the 12 MHz low gain crystal oscillator on the
+board with the on-chip PLL to generate a 180 MHz system clock.
+
+Serial Port
+===========
+
+The K66F SoC has six UARTs. None is used.
```

Comment:
How do you get a console? Segger RTT?

Reply:
@MaureenHelm Yes, console is only possible with Segger RTT with below settings as there is no hardware UART exposed for console or other purposes.

Device name. Default: MK66FN2M0XXX18 > 
Target interface. > SWD
Interface speed [kHz]. Default: 4000 kHz > 
RTT Control Block address. Default: auto-detection > 
RTT Channel name or index. Default: channel 1 > 0

Reply:
then add `CONFIG_USE_SEGGER_RTT=y` to `boards/arm/ip_k66f/ip_k66f_defconfig`

Reply:
@MaureenHelm Done, thanks